GOFevaluation/utils.py

```diff
@@ -359,24 +349,34 @@ def plot_equiprobable_histogram(data_sample, bin_edges, order=None,
     ax.set_xlim(xlim)
     ax.set_ylim(ylim)
 
+    plot_mode = 'sigma_deviation'
+    try:
+        plot_mode = kwargs.pop('plot_mode')
+    except KeyError:
+        pass
+
     ns = apply_irregular_binning(data_sample, bin_edges, order=order)
     ns = np.array(ns, dtype=float)
 
-    be = limit_bin_edges(bin_edges, data_sample, order)
+    be = get_finite_bin_edges(bin_edges, data_sample, order)
     be_first = be[0]
     be_second = be[1]
 
-    # get colormap and norm for colorbar
-    cmap_str = kwargs.get('cmap', 'RdBu_r')
-    cmap = mpl.cm.get_cmap(cmap_str)
-
-    if(plot_mode == 'count_density'):
-        ns = density_modifier(ns, be_first, be_second, data_sample)
+    if(plot_mode == 'sigma_deviation'):
+        cmap_str = kwargs.get('cmap', 'RdBu_r')
+        cmap = mpl.cm.get_cmap(cmap_str)
+        if nevents_expected is None:
+            raise ValueError('nevents_expected cannot ' +
+                             'be None while plot_mode=\'sigma_deviation\'')
+    elif(plot_mode == 'count_density'):
+        ns = get_count_density(ns, be_first, be_second, data_sample)
         cmap_str = kwargs.get('cmap', 'viridis')
         cmap = mpl.cm.get_cmap(cmap_str)
-    if(plot_mode == 'num_counts'):
+    elif(plot_mode == 'num_counts'):
         cmap_str = kwargs.get('cmap', 'viridis')
         cmap = mpl.cm.get_cmap(cmap_str)
+    else:
+        raise ValueError(f'plot_mode {plot_mode} is not defined.')
```

Comment:
I really like this solution! Now that we're already here, let's also put line 386 - 401 in the same if else structure, then we only need to iterate through the cases once and get all values that are relevant for the plots at one place.